I'm getting a weird error I'm not able to grok: I committed the wrong test file, which failed a build, which got me to checking the fine print. When I run `west build -p -b native_sim tests/posix/headers -T portability.posix.headers.picolibc.without_posix_api`, it causes a build error with `errno.h` not being able to find `asm/errno.h` - what could be the possible reason?

```
FAILED: zephyr/CMakeFiles/native_runner_executable zephyr/zephyr.exe /home/<USER>/zephyrproject/zephyr-local/build/zephyr/CMakeFiles/native_runner_executable /home/<USER>/zephyrproject/zephyr-local/build/zephyr/zephyr.exe 
cd /home/<USER>/zephyrproject/zephyr-local/build/zephyr && /usr/bin/make -f /home/<USER>/zephyrproject/zephyr-local/scripts/native_simulator/Makefile all --warn-undefined-variables -r NSI_CONFIG_FILE=/home/<USER>/zephyrproject/zephyr-local/build/zephyr/NSI/nsi_config
In file included from /usr/include/bits/errno.h:26,
                 from /usr/include/errno.h:28,
                 from /home/<USER>/zephyrproject/zephyr-local/drivers/entropy/fake_entropy_native_bottom.c:12:
/usr/include/linux/errno.h:1:10: fatal error: asm/errno.h: No such file or directory
    1 | #include <asm/errno.h>
      |          ^~~~~~~~~~~~~
```

--- a/include/zephyr/posix/stropts.h
+++ b/include/zephyr/posix/stropts.h
@@ -5,6 +5,7 @@
  */
 #ifndef ZEPHYR_INCLUDE_POSIX_STROPTS_H_
 #define ZEPHYR_INCLUDE_POSIX_STROPTS_H_
+#define RS_HIPRI BIT(0)
 
 #ifdef __cplusplus
 extern "C" {

--- a/include/zephyr/posix/stropts.h
+++ b/include/zephyr/posix/stropts.h
@@ -1,0 +1,25 @@
+/*
+ * Copyright (c) 2024 Abhinav Srivastava
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_POSIX_STROPTS_H_
+#define ZEPHYR_INCLUDE_POSIX_STROPTS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct strbuf {
+	int maxlen;
+	int len;
+	char *buf;
+};
+
+int putmsg(int fildes, const struct strbuf *ctlptr, const struct strbuf *dataptr, int flags);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* ZEPHYR_INCLUDE_POSIX_STROPTS_H_ */

--- a/lib/posix/CMakeLists.txt
+++ b/lib/posix/CMakeLists.txt
@@ -57,7 +57,7 @@ zephyr_library_sources_ifdef(CONFIG_PTHREAD_RWLOCK rwlock.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_PRIORITY_SCHEDULING sched.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC semaphore.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_SPINLOCK spinlock.c)
-zephyr_library_sources_ifdef(CONFIG_TIMER timer.c)
+zephyr_library_sources_ifdef(CONFIG_POSIX_PUTMSG stropts.c)
 
 zephyr_library_include_directories(
   ${ZEPHYR_BASE}/kernel/include

--- a/lib/posix/CMakeLists.txt
+++ b/lib/posix/CMakeLists.txt
@@ -57,6 +57,7 @@ zephyr_library_sources_ifdef(CONFIG_PTHREAD_RWLOCK rwlock.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_PRIORITY_SCHEDULING sched.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC semaphore.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_SPINLOCK spinlock.c)
+zephyr_library_sources_ifdef(CONFIG_TIMER timer.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_PUTMSG stropts.c)
 
 zephyr_library_include_directories(

--- a/lib/posix/Kconfig
+++ b/lib/posix/Kconfig
@@ -58,6 +58,7 @@ source "lib/posix/Kconfig.signal"
 source "lib/posix/Kconfig.spinlock"
 source "lib/posix/Kconfig.timer"
 source "lib/posix/Kconfig.uname"
+source "lib/posix/Kconfig.stropts"
 
 rsource "shell/Kconfig"
 

--- a/lib/posix/Kconfig.stropts
+++ b/lib/posix/Kconfig.stropts
@@ -1,0 +1,9 @@
+# copyright (c) 2024 Abhinav Srivastava
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config POSIX_PUTMSG
+	bool "Support for putmsg function"
+	default y if POSIX_API
+	help
+	  This option provides support for the putmsg function used in message passing.

--- a/lib/posix/stropts.c
+++ b/lib/posix/stropts.c
@@ -1,0 +1,20 @@
+/*
+ * Copyright (c) 2024 Abhinav Srivastava
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/posix/stropts.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+
+int putmsg(int fildes, const struct strbuf *ctlptr, const struct strbuf *dataptr, int flags)
+{
+	ARG_UNUSED(fildes);
+	ARG_UNUSED(ctlptr);
+	ARG_UNUSED(dataptr);
+	ARG_UNUSED(flags);
+
+	errno = ENOSYS;
+	return -1;
+}

--- a/tests/posix/common/src/stropts.c
+++ b/tests/posix/common/src/stropts.c
@@ -1,0 +1,21 @@
+/*
+ * Copyright (c) 2024 Abhinav Srivastava
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <stropts.h>
+#include <errno.h>
+
+ZTEST(stropts, test_putmsg)
+{
+	const struct strbuf *ctrl = NULL;
+	const struct strbuf *data = NULL;
+	int fd = -1;
+	int ret = putmsg(fd, ctrl, data, 0);
+
+	zassert_equal(ret, -1, "Expected return value -1, got %d", ret);
+	zassert_equal(errno, ENOSYS, "Expected errno ENOSYS, got %d", errno);
+}
+
+ZTEST_SUITE(stropts, NULL, NULL, NULL, NULL, NULL);

--- a/tests/posix/headers/src/stropts_h.c
+++ b/tests/posix/headers/src/stropts_h.c
@@ -1,0 +1,28 @@
+/*
+ * Copyright (c) 2024 Abhinav Srivastava
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "_common.h"
+#ifdef CONFIG_POSIX_API
+#include <stropts.h>
+#else
+#include <zephyr/posix/stropts.h>
+#endif
+
+/**
+ * @brief Test existence and basic functionality of stropts.h
+ *
+ * @see stropts.h
+ */
+ZTEST(posix_headers, test_stropts_h)
+{
+	#ifdef CONFIG_POSIX_API
+	zassert_not_null((void *)putmsg, "putmsg is null");
+
+	zassert_true(sizeof(((struct strbuf *)0)->maxlen) > 0, "maxlen size is 0");
+	zassert_true(sizeof(((struct strbuf *)0)->len) > 0, "len size is 0");
+	zassert_true(sizeof(((struct strbuf *)0)->buf) > 0, "buf size is 0");
+	zassert_not_equal(RS_HIPRI, ~RS_HIPRI);
+	#endif
+}

